server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java

```diff
@@ -39,10 +43,36 @@ public interface ChunkedRestResponseBody {
     Logger logger = LogManager.getLogger(ChunkedRestResponseBody.class);
```

Comment:
I wonder if we should change the name of this interface to `ChunkedRestResponseBodyPart`? That would make `isDone` less ambiguous (naming-wise). We could also rename `isDone` to `isPartDone` to make this more explicit.

I think the javadoc above could also be improved to mention that it can be multi-part.

Reply:
++ I will do this in a follow-up to avoid introducing too much noise here. I pushed 4570c922b1a to improve the Javadocs here tho.